test/cpp/jit/test_autodiff.cpp

```diff
@@ -28,8 +28,8 @@ using var_meta_list = std::vector<var_meta_type>;
 using test_fn_type = std::function<variable_list(const variable_list&)>;
 
 struct ADTestSpec {
-  ADTestSpec(const char* name, var_meta_list input_meta, test_fn_type test_fn)
-      : name(name), input_meta(input_meta), test_fn(test_fn) {}
+  ADTestSpec(const char* name, var_meta_list input_meta, test_fn_type test_fn, bool clamp = false)
+      : name(name), input_meta(input_meta), test_fn(test_fn), clamp(clamp) {}
```

Comment:
This doesn't seem very generic, since it's specifically clamping to 3f but the boolean just says "clamp in some way"

Reply:
Sure, let me change it to `float clampMax =-1` and skip clamping if value is negative